test/e2e/kappcontroller/template_test.go

```diff
@@ -164,3 +164,89 @@ from_path: true
 		}
 	})
 }
+
+func Test_SecretsAndConfigMapsWithCustomPathsCanReconcile(t *testing.T) {
+	env := e2e.BuildEnv(t)
+	logger := e2e.Logger{}
+	kapp := e2e.Kapp{t, env.Namespace, logger}
+	kubectl := e2e.Kubectl{t, env.Namespace, logger}
+	sas := e2e.ServiceAccounts{env.Namespace}
+
+	name := "inline-pathsfrom-directorypath"
+	appYaml := `---
+apiVersion: kappctrl.k14s.io/v1alpha1
+kind: App
+metadata:
+  name: simple-app2
+  namespace: kappctrl-test
+  annotations:
+    kapp.k14s.io/change-group: kappctrl-e2e.k14s.io/apps
+spec:
+  serviceAccountName: kappctrl-e2e-ns-sa
+  fetch:
+  - inline:
+      paths:
+        file-from-fetch-paths: ""
+        template/dir/blah.txt: ""
+      pathsFrom:
+        - secretRef:
+            name: from-secret
+            directoryPath: fetch/dis/dir
+  template:
+  - ytt:
+      inline:
+        pathsFrom:
+        - secretRef:
+            name: from-secret
+            directoryPath: template/dir
+        paths:
+          config.yml: |
+            #@ load("@ytt:data", "data")
+            #@ load("@ytt:yaml", "yaml")
+
+            #@ print(data.list())
+
+            kind: ConfigMap
+            apiVersion: v1
+            metadata:
+              name: foo
+            data:
+              data: #@ yaml.encode(data.list())
+  deploy:
+  - kapp: {}
+---
+kind: Secret
+apiVersion: v1
+metadata:
+  name: from-secret
+  namespace: kappctrl-test
+stringData:
+  file-from-secret: "thisIsData"
```

Comment:
add a second secret that also has the `file-from-secret` key,   put that secret in a separate directoryPath, then at least the test is more motivated.